src/pathOr.js

```diff
@@ -0,0 +1,30 @@
+var _curry3 = require('./internal/_curry3');
+
+
+/**
+ * If the given, non-null object has an own property at the given path, returns
+ * the value at that path. Otherwise returns the provided default value.
+ *
+ * @func
+ * @memberOf R
+ * @category Object
+ * @sig a -> [String] -> {k: v} -> v | a
+ * @param {*} d The default value.
+ * @param {Array} path The path to use.
+ * @return {*} The data at `path`.
+ * @example
+ *
+ *      R.pathOr('N/A', ['a', 'b'], {a: {b: 2}}); //=> 2
+ *      R.pathOr('N/A', ['a', 'b'], {c: {b: 2}}); //=> "N/A"
+ */
+module.exports = _curry3(function path(d, paths, obj) {
+  if (obj == null) {
+    return d;
+  } else {
+    var val = obj;
+    for (var idx = 0, len = paths.length; idx < len && val != null; idx += 1) {
+      val = val[paths[idx]];
+    }
```

Comment:
Ramda style favours `while` loops over `for` loops. Also, we prefer not to cache array lengths unless necessary. We might write:

```javascript
var val = obj;
var idx = 0;
while (val != null && idx < path.length) {
  val = val[path[idx]];
  idx += 1;
}
```